Solutions/Corvus.Storage.Azure.Cosmos/Microsoft/Extensions/DependencyInjection/CosmosServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// <copyright file="CosmosServiceCollectionExtensions.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+using Corvus.Storage.Azure.Cosmos;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Common configuration code for services using Azure Cosmos DB via SQL.
+    /// </summary>
+    public static class CosmosServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds an <see cref="ICosmosContainerSourceByConfiguration"/> to the service collection.
```

Comment:
The type it registers is the `ByConfiguration` version, but the method name does not correspond to that. Is that intentional?

Reply:
It is deliberate, but on reflection it's probably not very good.

The reason for avoiding `ByConfiguration` is that when you put that in a DI extension method, it ends up looking a lot like it's saying it's going to load up some app-level configuration in the `IConfiguration` sense, because that's most often the source of configuration you're dealing with during DI setup.

This is the same issue that led to me using the `...FromDynamicConfiguration` naming in `Corvus.Identity`. And I'm now wondering whether I should just use the same thing here, so the blob and Cosmos registration methods would look like:

```cs
services.AddCosmosContainerSourceFromDynamicConfiguration()
services.AddBlobContainerSourceFromDynamicConfiguration()
```

and I think it would make sense to rename the relevant interfaces too, so `ICosmosContainerSourceByConfiguration` would become `ICosmosContainerSourceFromDynamicConfiguration`, and `IBlobContainerSourceByConfiguration` would become IBlobContainerSourceFromDynamicConfiguration`.

What do you think.

Reply:
Having slept on this I've now pushed a change that brings the naming in here in line with the naming in `Corvus.Identity`: I'm using this `FromDynamicConfiguration` naming throughout.